packages/grafana-ui/src/components/Legend/Legend.tsx

```diff
@@ -0,0 +1,50 @@
+import React from 'react';
+
+import { StatID } from '../../utils/statsCalculator';
+
+export interface LegendBasicOptions {
+  isVisible: boolean;
+  asTable: boolean;
+}
+
+export interface LegendRenderOptions {
+  placement: LegendPlacement;
+  hideEmpty?: boolean;
+  hideZero?: boolean;
+}
+
+export type LegendPlacement = 'under' | 'right' | 'over'; // Over used by piechart
+
+export interface LegendOptions extends LegendBasicOptions, LegendRenderOptions {
+  stats?: StatID[];
+  decimals?: number;
+}
+
+export interface LegendItem {
+  label: string;
+  color: string;
+  isVisible: boolean;
+  stats: Array<{ statId: string; value: number }>;
```

Comment:
On my way to do that!

Reply:
@ryantxu I have introduced `StatsDisplayValue` type that extends `DisplayValue` by adding required `statId: string` property. It felt unnatural to rely on any prop from `DisplayValue` to use it as `id` which will be necessary for sorting by given stat.

Reply:
hymmm -- can you you just use the `title` property instead?

I am not wild about linking stats to the legend component because (as a library) the component should not care where the info came from.

Reply:
Yeah, I'm gonna change that. The only place I think `stats` (as `DisplayValue` though) will be present is in the LegendItem interface, and that's because of the nature of legend implementations that now vary from simple string s list without any possibility of manipulation, to sortable table. The only thought I have is whether or not we should call it `stats`. Now we use stats in Legend, but I assume we could provide the legend item with any sort of additional DisplayValues that any abstract legend component could use. Hence I'm thinking about replacing `stats` with either `seriesMetadata` or simply `values`. WDYT @ryantxu ?

Reply:
Or just `info` as you called it `GraphSeriesXY`

Reply:
Then using `title` as id for i.e. sorting would be fine. The only thing I think will complicate the implementation is the fact that `title` is optional property.

Reply:
I like `info`.

IIUC, you really just need `title` for a consistent way to sort in the table format.  I vote you either use something like:
```
export interface LegendDisplayValue extends DisplayValue {
  key: string;
}
```

Or (even better) can you have your key/column title be based on either the `title` or the index?  

Keeping title optional is nice so that the display does not force: 'Min: XXX', 'Max: YYY'